radix-engine/src/model/epoch_manager/executables.rs

```diff
@@ -202,17 +219,25 @@ impl Executor for EpochManagerNextRoundExecutable {
         }
 
         if self.round >= epoch_manager.rounds_per_epoch {
-            let next_epoch = epoch_manager.epoch + 1;
-            epoch_manager.epoch = next_epoch;
+            let offset = SubstateOffset::EpochManager(EpochManagerOffset::PreparingValidatorSet);
+            let handle = system_api.lock_substate(self.node_id, offset, LockFlags::MUTABLE)?;
+            let mut substate_mut = system_api.get_ref_mut(handle)?;
+            let preparing_validator_set = substate_mut.validator_set();
+            let prepared_epoch = preparing_validator_set.epoch;
+            let next_validator_set = preparing_validator_set.validator_set.clone();
+            preparing_validator_set.epoch = prepared_epoch + 1;
```

Comment:
We do not update `preparing_validator_set`?

Reply:
Nope, we shouldn't need to. `preparing_validator_set` gets updated in the middle of epochs rather than the end.